### PR TITLE
[WIP]: add "open" option to node api to allow opening in browser

### PR DESCRIPTION
**Pull request by amilajack**
- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### Motivation / Use-Case

Node api does not support open option

### Additional Info

### CODE REVIEW

**Comment by amilajack** [comment]
Does this conflict with https://github.com/webpack/webpack-dev-server/pull/1540/files?

File: bin/utils.js [hunk 1]
```diff
@@ -70,22 +69,6 @@ function status (uri, options, log, useColor) {
...
-      openOptions = { app: options.open };
-      openMessage += `: ${options.open}`;
-    }
-
-    open(uri + (options.openPage || ''), openOptions).catch(() => {
-      log.warn(
-        `${openMessage}. If you are running in a headless environment, please do not use the --open flag`
-      );
-    });
-  }
```
Comment (evilebottnawi) [inline bin/utils.js]: Don't move this from utils

File: lib/Server.js [hunk 2]
```diff
@@ -804,6 +825,12 @@ Server.prototype.listen = function (port, hostname, fn) {
     if (fn) {
       fn.call(this.listeningApp, err);
     }
+
+    if (this.options.open) {
+      const suffix = (this.options.inline !== false || this.options.lazy === true ? '/' : '/webpack-dev-server/');
+      const uri = createDomain({ host: hostname, port }, this.listeningApp) + suffix;
+      openBrowser(uri, this.options, this.log);
```
Comment (evilebottnawi) [inline lib/Server.js]: It is should be in listen event and remove logic for open from bin file

**Comment by amilajack** [comment]
@evilebottnawi @michael-ciniawsky it seems like https://github.com/webpack/webpack-dev-server/pull/1540 does the same thing as this PR. Should this PR be closed in favor of that PR?

**Comment by evilebottnawi** [comment]
@amilajack PR was abadoned, so you can continue develop

File: lib/Server.js (hunk 2 above)
Comment (amilajack) [reply lib/Server.js]: > remove logic for open from bin file

I'm not sure what do you mean by this. The previous comment says that open logic should be kept in bin

Comment (amilajack) [reply lib/Server.js]: Also socksjs doesn't seem to register a 'listen' event. On what object should i listen for and where should i call it?

Comment (evilebottnawi) [reply lib/Server.js]: @amilajack listen `express` events https://github.com/webpack/webpack-dev-server/blob/master/bin/webpack-dev-server.js#L408

Comment (amilajack) [reply lib/Server.js]: But if I listen with express what port should i listen on? Also i don't understand the reason for listening on another port with express. Can you please explain this.

**Comment by hiroppy** [comment]
friendly ping @amilajack

**Pull request closed without merging**